avro_schema/frontend.lua

```diff
@@ -135,6 +135,24 @@ local copy_schema
 local copy_schema_error
 local copy_schema_location_info
 
+-- add new type definition to scope
+local function scope_add_type(scope, options, typeid, xtype)
```

Comment:
`scope` can be replaced with booking table from graphql with the following addition code:

* add booking table option to forbid non-empty value reassignment;
* add check that all used types are defined
* Alex said that nullability need some separate tricks.